scipy/io/mmio.py

```diff
@@ -505,7 +532,7 @@ def _parse_body(self, stream):
         return a
 
     #---------------------------------------------------------------------------
-    def _write(self, stream, a, comment='', field=None, precision=None):
+    def _write(self, stream, a, comment='', field=None, precision=None, symmetry=None):
```

Comment:
Before a possible symmetry was not taken into account and the entire matrix was always saved (in `general` format). The advantage of the `symmetry` formats is (almost) half disk usage.
The symmetry check can be slow for large matrices. If possible, the user should therefore pass the symmetry type to the `mmwrite` method.